runtime/arc.js

```diff
@@ -159,7 +169,30 @@ class Arc {
     return s;
   }
 
-  connectParticleToView(particleHandle, particle, name, targetView) {
+  instantiate(recipe) {
+    assert(recipe.isResolved(), 'Cannot instantiate an unresolved recipe');
+    let {views, particles, slots} = recipe.mergeInfo(this._activeRecipe);
+    let recipeViewMap = new Map();
+    views.forEach(recipeView => {
+      let view;
+      if (recipeView.create) {
+        view = this.createView(recipeView.type);
```

Comment:
after creating a view, we also need to copy its ID back into the active recipe and remove the create flag.